src/HttpRequestDurationsMiddlewareBuilderExtensions.cs

```diff
@@ -29,6 +29,12 @@ public static IApplicationBuilder UsePrometheusRequestDurations(this IApplicatio
                 ??= (ICollectorRegistry)app.ApplicationServices.GetService(typeof(ICollectorRegistry))
                     ?? Metrics.DefaultCollectorRegistry;
 
+#if HasRoutes
+            // If we are going to set labels for Controller or Action -- then we need to make them readily available
+            if (options.IncludeController || options.IncludeAction)
+                app.UseMiddleware<CaptureRouteDataMiddleware>();
```

Comment:
Can you extract logic from `CaptureRouteDataMiddleware` and add to `HttpRequestDurationsMiddleware`?

Reply:
Maybe I need some details about `CaptureRouteDataMiddleware`, why it should be like this.

Reply:
I leveraged the same pattern to obtain and get `Controller` and `Action` added as labels as the prior prometheus-net sdk is doing.  Let me know if you want me to combine into the existing middleware...

Reply:
@billpieper Yes, I think that we need only 1 middleware `HttpRequestDurationsMiddleware`, some options can be enabled or disabled